lib/game.rb

```diff
@@ -1,29 +1,23 @@
 module TicTacToe
 
   class Game 
-    TTT_MARKER1 = "X"
-    TTT_MARKER2 = "O"
-    
-    def initialize(game_turn, marker_loop, game_completion, game_eval)
+    attr_reader :player1
+    attr_reader :player2
+
+    def initialize(game_turn, game_eval, p1_marker, p2_marker)
       @game_turn = game_turn
-      @marker_loop = marker_loop
-      @game_completion = game_completion
       @game_eval = game_eval
+      @player1 = Player.new(1, p1_marker)
+      @player2 = Player.new(2, p2_marker)
     end
 
-    def run 
-      marker1, marker2 = @marker_loop.marker_selection(TTT_MARKER1, TTT_MARKER2)
-
-      @player1, @player2 = Player.new(1, marker1), Player.new(2, marker2)
-
+    def players_turns
       current_player = @player1
-
       begin 
         board_array = @game_turn.execute(current_player.ord_num, current_player.marker)
         current_player = toggle_player(current_player)
-      end until @game_eval.game_over?(board_array, TTT_MARKER1, TTT_MARKER2)
-      
-      @game_completion.game_over_messages(board_array, @player1, @player2)
+      end until @game_eval.game_over?(board_array, @player1, @player2)
```

Comment:
I like this idea, and I'm happy to make this change. But couldn't a similar argument be said of the #player_won method in GameEval? The method heavily relies on the implementation of the Player object. I definitely agree with the reasons we have of being skeptical of potential change to Player, but by changing line 19 I still won't completely get rid of the dependency that GameEval has on Player (although I'll be able to reduce it)